src/main/java/seedu/address/logic/parser/AddRecordCommandParser.java

```diff
@@ -23,17 +27,18 @@ public class AddRecordCommandParser implements Parser<AddRecordCommand> {
      */
     public AddRecordCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_DATE, PREFIX_RECORD);
+                ArgumentTokenizer.tokenize(args, PREFIX_DATE, PREFIX_RECORD, PREFIX_MEDICATION);
 
         if (!arePrefixesPresent(argMultimap, PREFIX_DATE, PREFIX_RECORD)
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddRecordCommand.MESSAGE_USAGE));
         }
 
-        String recordDate = argMultimap.getValue(PREFIX_DATE).get();
-        String recordData = argMultimap.getValue(PREFIX_RECORD).get();
+        LocalDateTime recordDate = ParserUtil.parseDate(argMultimap.getValue(PREFIX_DATE).get());
+        String recordData = ParserUtil.parseRecordData(argMultimap.getValue(PREFIX_RECORD).get());
+        Set<Medication> medications = ParserUtil.parseMedications(argMultimap.getAllValues(PREFIX_MEDICATION));
```

Comment:
Perhaps for the parsing of medication, we can use the "," delimiter to split each individual medication for brevity.

Reply:
Do you mean something like parseMedication("X, Y, Z") ? I think that's a possible solution but might not be as user-friendly in cases where the user wants to input a medication and dosage separately (eg. Paracetamol, 500mg). The current explicit use of the prefix "m/" follows the implementation of `Tag`.

I'm open to opinions from the rest also. Might consider adding a field in `Medication` explicitly for `Dosage`.

Reply:
I feel using ',' might be better since it makes it faster to type compared to /m and would benefit doctors. This is especially so if the patient is prescribed many types of medicine at once. Perhaps ':' could be used to seperate the medication and the dosage
(eg. m/Paracetamol: 500mg, dextromethorphan: 8ml)